ls1intum/Artemis
# [Bugfix] - Save Complaint after Instructor overwrote assessment + Server & Client Tests

Fixes https://github.com/ls1intum/Artemis/issues/2026
Fixes https://github.com/ls1intum/Artemis/issues/2129



### Checklist

### Motivation and Context


- Weekly client side test coverage improvements
- When overwriting an assessment as instructor after a complaint was created for this submission, the complaint is not shown anymore

### Description

- Get the client test coverage higher 🚀 
- Integration test covering the problem mentioned in https://github.com/ls1intum/Artemis/issues/2026 and https://github.com/ls1intum/Artemis/issues/2129 for modeling exercises and for programming exercises.
- Fixed a Bug where the complaint is overwritten in programming exercises
- Added some more delay on one test which did not pass once (Should fail less often since the test/call has more time to be executed)

### Steps for Testing


1. Log in to Artemis
2. Create a Programming-Exercise
3. Participate
4. Assess
5. Complain
6. (Accept/Reject the complaint) - Step not necessarily needed
7. As a instructor, override the assessment
8. The complaint should still be visible in the assessment view (same view when clicking in _override Assessment_)

### Test Coverage



short-answer-question.component.ts: 0% -> 94%
